tests/integration_tests/security/test_vulnerabilities.py

```diff
@@ -115,141 +184,129 @@ def test_spectre_meltdown_checker_on_host(spectre_meltdown_checker):
     )
 
 
-@pytest.mark.no_block_pr
 @pytest.mark.skipif(
     global_props.instance == "c7g.metal" and global_props.host_linux_version == "4.14",
     reason="c7g host 4.14 requires modifications to the 5.10 guest kernel to boot successfully.",
 )
-def test_spectre_meltdown_checker_on_guest(spectre_meltdown_checker, microvm):
+def test_spectre_meltdown_checker_on_guest(spectre_meltdown_checker, build_microvm):
     """
     Test with the spectre / meltdown checker on guest.
     """
-    run_spectre_meltdown_checker_on_guest(
-        microvm,
-        spectre_meltdown_checker,
+
+    git_ab_test_guest_command_if_pr(
+        with_checker(build_microvm, spectre_meltdown_checker),
+        REMOTE_CHECKER_COMMAND,
+        comparator=set_did_not_grow_comparator(
+            spectre_meltdown_reported_vulnerablities
+        ),
     )
 
 
-@pytest.mark.no_block_pr
 @pytest.mark.skipif(
     global_props.instance == "c7g.metal" and global_props.host_linux_version == "4.14",
     reason="c7g host 4.14 requires modifications to the 5.10 guest kernel to boot successfully.",
 )
 def test_spectre_meltdown_checker_on_restored_guest(
-    spectre_meltdown_checker,
-    microvm,
-    microvm_factory,
+    spectre_meltdown_checker, build_microvm, microvm_factory
 ):
     """
     Test with the spectre / meltdown checker on a restored guest.
     """
-
-    snapshot = microvm.snapshot_full()
-    # Create a destination VM
-    dst_vm = microvm_factory.build()
-    dst_vm.spawn()
-    # Restore the destination VM from the snapshot
-    dst_vm.restore_from_snapshot(snapshot, resume=True)
-
-    run_spectre_meltdown_checker_on_guest(
-        dst_vm,
-        spectre_meltdown_checker,
+    git_ab_test_guest_command_if_pr(
+        with_checker(
+            with_restore(build_microvm, microvm_factory), spectre_meltdown_checker
+        ),
```

Comment:
I think it would be easier to read if it was:

```python
decorators = [build_microvm, restore, partial(check, check=spectre_meltdown_checker)]
for decorator in decorators:
    res = decorator(res)
```

But I am not sure if it that would allow passing the firecracker binaries. 

Could we turn it inside out?

```
@pytest.fixture
def microvm_factory_a: pass

@pytest.fixture
def microvm_factory_b: pass

def test_ab(microvm_factory_a, microvm_factory_b):
    assert _test(microvm_factory_a) == _test(microvm_factory_b)
```

Reply:
The problem with having a `microvm_factory_b` fixture is deciding what should happen if the test is not run on a PR (e.g. in a nightly run where we don't want to do an A/B test and instead want to fail if the command fails on the HEAD of main) :(

Reply:
True. I think we can always simplify it later if we figure out a way.